diff --git a/Testing/Economic.ApplicationService.Tests/CustomerApplicationServiceTests.cs b/Testing/Economic.ApplicationService.Tests/CustomerApplicationServiceTests.cs
new file mode 100644
index 0000000000..b98ab94b07
--- /dev/null
+++ b/Testing/Economic.ApplicationService.Tests/CustomerApplicationServiceTests.cs
@@ -0,0 +1,171 @@
+using System;
+using System.Linq;
+using Economic.Entities;
+using Economic.Infrastructure.Interface;
+using Economic.RepositoryService.Interface;
+using Moq;
+using NUnit.Framework;
+
+namespace Economic.ApplicationService.Tests
+{
+	public class CustomerApplicationServiceTests
+	{
+		private CustomerApplicationService _sut;
+
+		[OneTimeSetUp]
+		public void SetUp()
+		{
+			_sut = new CustomerApplicationService(Mock.Of<IExecutionContext>(), 
+				Mock.Of<ICustomerRepositoryService>());
+		}
+		
+		[Test]
+		public void Get_all_customer_identifiers()
+		{
+			// arrange
+			var cvr = "cvr_number";
+			var pnumber = "p_number";
+			var ean = "ean_number";
+			
+			Customer customer = new Customer(0,
+				0,
+				0,
+				0,
+				"",
+				false,
+				DateTime.Now,
+				string.Empty,
+				string.Empty,
+				"",
+				"",
+				"",
+				"",
+				"",
+				"",
+				"",
+				0,
+				"",
+				cvr,
+				0,
+				0,
+				0,
+				"",
+				0,
+				0,
+				0,
+				0,
+				0,
+				0,
+				0,
+				ean,
+				"",
+				"",
+				"",
+				decimal.Zero,
+				"",
+				decimal.Zero,
+				"",
+				"",
+				"",
+				DateTime.Now,
+				pnumber,
+				0,
+				"",
+				"",
+				"",
+				"");
+			
+			// act
+			var identifiers = _sut.GetIdentifiers(customer).ToList();
+			
+			// assert
+			Assert.IsNotNull(identifiers);
+			Assert.IsNotEmpty(identifiers);
+			Assert.That(identifiers.Count(), Is.EqualTo(3));
+		}
+		
+		[Test]
+		public void Get_all_changed_identifiers()
+		{
+			// arrange
+			var orginalIdentifier = "orginal_identifier";
+			var cvr = "cvr_number";
+			var pnumber = "p_number";
+			var ean = "ean_number";
+			
+			Customer customer = new Customer(0,
+				0,
+				0,
+				0,
+				"",
+				false,
+				DateTime.Now,
+				string.Empty,
+				string.Empty,
+				"",
+				"",
+				"",
+				"",
+				"",
+				"",
+				"",
+				0,
+				"",
+				orginalIdentifier,
+				0,
+				0,
+				0,
+				"",
+				0,
+				0,
+				0,
+				0,
+				0,
+				0,
+				0,
+				orginalIdentifier,
+				"",
+				"",
+				"",
+				decimal.Zero,
+				"",
+				decimal.Zero,
+				"",
+				"",
+				"",
+				DateTime.Now,
+				orginalIdentifier,
+				0,
+				"",
+				"",
+				"",
+				"");
+
+			customer.PNr = pnumber;
+			customer.EANLocationNumber = ean;
+			customer.CvrNo = cvr;
+			
+			// act
+			var identifiers = _sut.GetIdentifiers(customer).ToList();
+			
+			// assert
+			Assert.IsNotNull(identifiers);
+			Assert.IsNotEmpty(identifiers);
+			Assert.That(identifiers.Count(), Is.EqualTo(3));
+		}
+		
+		[Test]
+		public void Null_customer_returns_empty_identifiers()
+		{
+			// arrange
+			Customer customer = null;
+			
+			// act
+			var identifiers = _sut.GetIdentifiers(customer);
+			
+			// assert
+			Assert.NotNull(identifiers);
+			Assert.That(identifiers.Count(), Is.Zero);
+		}
+	}
+}
diff --git a/Testing/Economic.Service.Test/SproomCompanyIdentifierTests.cs b/Testing/Economic.Service.Test/SproomCompanyIdentifierTests.cs
index e93e01b754..547606ad90 100644
--- a/Testing/Economic.Service.Test/SproomCompanyIdentifierTests.cs
+++ b/Testing/Economic.Service.Test/SproomCompanyIdentifierTests.cs
@@ -1,4 +1,4 @@
-using Economic.Services.Sproom;
+using Economic.Dto.Sproom;
 using NUnit.Framework;
 
 namespace Economic.Service.Tests
diff --git a/code/Economic.ApplicationService.Interface/ICustomerApplicationService.cs b/code/Economic.ApplicationService.Interface/ICustomerApplicationService.cs
index c5b8121fb8..800c941e2f 100644
--- a/code/Economic.ApplicationService.Interface/ICustomerApplicationService.cs
+++ b/code/Economic.ApplicationService.Interface/ICustomerApplicationService.cs
@@ -1,5 +1,6 @@
 ﻿using System.Collections.Generic;
 using Economic.Dto;
+using Economic.Entities;
 
 namespace Economic.ApplicationService.Interface
 {
@@ -7,5 +8,7 @@ public interface ICustomerApplicationService
 	{
 		IEnumerable<CustomerDto> CustomersWithCvrNumber(IEconMainSqlConnection connection, int setupId, string cvrNumber);
 		CustomerDto Get(int customerId);
+		IEnumerable<string> GetChangedIdentifiers(Customer customer);
+		IEnumerable<string> GetIdentifiers(Customer customer);
 	}
 }
diff --git a/code/Economic.ApplicationService/CustomerApplicationService.cs b/code/Economic.ApplicationService/CustomerApplicationService.cs
index 0749c14815..5159b99054 100644
--- a/code/Economic.ApplicationService/CustomerApplicationService.cs
+++ b/code/Economic.ApplicationService/CustomerApplicationService.cs
@@ -1,6 +1,8 @@
 ﻿using System.Collections.Generic;
 using Economic.ApplicationService.Interface;
 using Economic.Dto;
+using Economic.Dto.Sproom;
+using Economic.Entities;
 using Economic.Infrastructure.Interface;
 using Economic.RepositoryService.Interface;
 
@@ -11,13 +13,15 @@ public class CustomerApplicationService : ApplicationServiceBase, ICustomerAppli
 		private readonly ICustomerRepositoryService _customerRepositoryService;
 		private readonly IExecutionContext _executionContext;
 
-		public CustomerApplicationService(IExecutionContext executionContext, ICustomerRepositoryService customerRepositoryService)
+		public CustomerApplicationService(IExecutionContext executionContext,
+			ICustomerRepositoryService customerRepositoryService)
 		{
 			_customerRepositoryService = customerRepositoryService;
 			_executionContext = executionContext;
 		}
 
-		public IEnumerable<CustomerDto> CustomersWithCvrNumber(IEconMainSqlConnection connection, int setupId, string cvrNumber)
+		public IEnumerable<CustomerDto> CustomersWithCvrNumber(IEconMainSqlConnection connection, int setupId,
+			string cvrNumber)
 		{
 			return _customerRepositoryService.GetByCvrNumber(connection, setupId, cvrNumber);
 		}
@@ -26,5 +30,58 @@ public CustomerDto Get(int customerId)
 		{
 			return _customerRepositoryService.Get(_executionContext.Connection, _executionContext.SetupId, customerId);
 		}
+
+		public IEnumerable<string> GetChangedIdentifiers(Customer customer)
+		{
+			var result = new List<string>();
+
+			if (customer.CvrNoHasChanged && !string.IsNullOrEmpty(customer.CvrNo))
+			{
+				result.Add(new SproomCompanyIdentifier(ElectronicInvoiceRecipientType.Cvr, customer.CvrNo)
+					.GetFormattedString());
+			}
+
+			if (customer.EANLocationNumberHasChanged && !string.IsNullOrEmpty(customer.EANLocationNumber))
+			{
+				result.Add(new SproomCompanyIdentifier(ElectronicInvoiceRecipientType.Gln, customer.EANLocationNumber)
+					.GetFormattedString());
+			}
+
+			if (customer.PNrHasChanged && !string.IsNullOrEmpty(customer.PNr))
+			{
+				result.Add(new SproomCompanyIdentifier(ElectronicInvoiceRecipientType.PNumber, customer.PNr)
+					.GetFormattedString());
+			}
+
+			return result;
+		}
+
+		public IEnumerable<string> GetIdentifiers(Customer customer)
+		{
+			var result = new List<string>();
+
+			if (customer == null)
+				return result;
+
+			if (!string.IsNullOrEmpty(customer.CvrNo))
+			{
+				result.Add(new SproomCompanyIdentifier(ElectronicInvoiceRecipientType.Cvr, customer.CvrNo)
+					.GetFormattedString());
+			}
+
+			if (!string.IsNullOrEmpty(customer.EANLocationNumber))
+			{
+				result.Add(new SproomCompanyIdentifier(ElectronicInvoiceRecipientType.Gln, customer.EANLocationNumber)
+					.GetFormattedString());
+			}
+
+			if (!string.IsNullOrEmpty(customer.PNr))
+			{
+				result.Add(new SproomCompanyIdentifier(ElectronicInvoiceRecipientType.PNumber, customer.PNr)
+					.GetFormattedString());
+			}
+
+			return result;
+		}
 	}
 }
diff --git a/code/r1/dotnet/src/Services/Sproom/SproomCompanyIdentifier.cs b/code/Economic.Dto/Sproom/SproomCompanyIdentifier.cs
similarity index 82%
rename from code/r1/dotnet/src/Services/Sproom/SproomCompanyIdentifier.cs
rename to code/Economic.Dto/Sproom/SproomCompanyIdentifier.cs
index b2f3a8431f..1a2c7e0e7f 100644
--- a/code/r1/dotnet/src/Services/Sproom/SproomCompanyIdentifier.cs
+++ b/code/Economic.Dto/Sproom/SproomCompanyIdentifier.cs
@@ -1,14 +1,14 @@
 using System;
 using System.Collections.Generic;
 
-namespace Economic.Services.Sproom
+namespace Economic.Dto.Sproom
 {
 	public class SproomCompanyIdentifier
 	{
 		private readonly ElectronicInvoiceRecipientType _type;
 		private readonly string _identifier;
 
-		private readonly Dictionary<ElectronicInvoiceRecipientType, Func<string, string>> _formatter =
+		private static readonly Dictionary<ElectronicInvoiceRecipientType, Func<string, string>> _formatter =
 			new Dictionary<ElectronicInvoiceRecipientType, Func<string, string>>
 			{
 				[ElectronicInvoiceRecipientType.Gln] = (identifier) => $"GLN:{identifier}",
diff --git a/code/Economic.Entity/Customer.cs b/code/Economic.Entity/Customer.cs
index 9741d0a162..c291b54123 100644
--- a/code/Economic.Entity/Customer.cs
+++ b/code/Economic.Entity/Customer.cs
@@ -185,6 +185,8 @@ public string BankAccount
 			set => bankAccount.Set(value);
 		}
 
+		public bool CvrNoHasChanged => cvrNo.OriginalValue != cvrNo.Value;
+		
 		[EntityColumnAttribute("CVR")]
 		public string CvrNo
 		{
@@ -268,6 +270,8 @@ public string Homepage
 			get => defaultDeliveryLocation.Value;
 			set => defaultDeliveryLocation.Set(value);
 		}
+		
+		public bool EANLocationNumberHasChanged => eanLocationNumber.OriginalValue != eanLocationNumber.Value;
 
 		[EntityColumnAttribute("EANLokationsNr")]
 		public string EANLocationNumber
@@ -346,6 +350,8 @@ public string SWIFT
 			set => directDebitMandateDate.Set(value);
 		}
 
+		public bool PNrHasChanged => pnr.OriginalValue != pnr.Value;
+		
 		[EntityColumnAttribute("PNr")]
 		public string PNr
 		{
diff --git a/code/r1/dotnet/AspCore/src/ClassicASP/ASPCore.cs b/code/r1/dotnet/AspCore/src/ClassicASP/ASPCore.cs
index 6c1992edae..3e66b5ac22 100644
--- a/code/r1/dotnet/AspCore/src/ClassicASP/ASPCore.cs
+++ b/code/r1/dotnet/AspCore/src/ClassicASP/ASPCore.cs
@@ -92,6 +92,7 @@
 using Economic.Services.Reminders;
 using Economic.Services.Reports.ChartOfAccountsReport;
 using Economic.Services.SalesDocuments;
+using Economic.Services.Sproom;
 using Economic.Services.Suppliers;
 using Economic.Services.User;
 using Economic.Services.YourPay;
@@ -1442,6 +1443,9 @@ public object GetPaymentIdFromSupplier(int setup, int supplierNumber)
 																		{"mobile-phone", mobilePhone},
 																	}));
 			RegisterWebHookEvent_CustomerUpdated(null, Convert.ToString(customerNumber));
+			
+			var identifiers = context.GetService<ICustomerApplicationService>().GetIdentifiers(customer);
+			context.GetService<ISproomService>().GetCompanyElectronicInvoiceCapability(identifiers);
 		}
 
 		public void UpdateCustomer(int setupId,
@@ -1492,6 +1496,7 @@ public object GetPaymentIdFromSupplier(int setup, int supplierNumber)
 									string defaultOrderEmail,
 									string defaultQuoteEmail)
 		{
+			var customerApplicationService = context.GetService<ICustomerApplicationService>();
 			var customerRepository = this.context.ServiceContainer.Get<EntityRepositories.ICustomerRepository>();
 			Customer existingCustomer = customerRepository.Get(this.context.Connection, setupId, customerNumber);
 			if (existingCustomer == null)
@@ -1549,6 +1554,9 @@ public object GetPaymentIdFromSupplier(int setup, int supplierNumber)
 										},
 										setupId);
 			RegisterWebHookEvent_CustomerUpdated(Convert.ToString(customerNumber), Convert.ToString(newCustomerNumber));
+
+			var changedIdentifiers = customerApplicationService.GetChangedIdentifiers(existingCustomer);
+			context.GetService<ISproomService>().GetCompanyElectronicInvoiceCapability(changedIdentifiers);
 		}
 
 		public void SetKladdePosteringBetaling(int KladdeNr,
diff --git a/code/r1/dotnet/src/Services/SalesDocuments/SalesDocumentService.cs b/code/r1/dotnet/src/Services/SalesDocuments/SalesDocumentService.cs
index 6fb0b0bf46..8fe235739f 100644
--- a/code/r1/dotnet/src/Services/SalesDocuments/SalesDocumentService.cs
+++ b/code/r1/dotnet/src/Services/SalesDocuments/SalesDocumentService.cs
@@ -3,13 +3,13 @@
 using System.Linq;
 using Economic.Constants;
 using Economic.Dto.SalesDocuments;
+using Economic.Dto.Sproom;
 using Economic.Entity;
 using Economic.EntityRepositories;
 using Economic.Infrastructure.Exceptions.Api;
 using Economic.Repository.SalesDocument;
 using Economic.Sql.Interface;
 using EconomicEnum;
-using Economic.Services.Sproom;
 
 namespace Economic.Service.SalesDocuments
 {
diff --git a/fix-f1-branch.patch b/fix-f1-branch.patch
new file mode 100644
index 0000000000..8d72869a74
--- /dev/null
+++ b/fix-f1-branch.patch
@@ -0,0 +1,39 @@
+diff --git a/rest/src/eco.rest.testutilities/TestEnvironment.cs b/rest/src/eco.rest.testutilities/TestEnvironment.cs
+index 62cff7b933..47f6b620c0 100644
+--- a/rest/src/eco.rest.testutilities/TestEnvironment.cs
++++ b/rest/src/eco.rest.testutilities/TestEnvironment.cs
+@@ -1,4 +1,5 @@
+ ∩╗┐using System.IO;
++using eco.rest.Infrastructure.Economic;
+ using Economic;
+ using Economic.Rest;
+ using Economic.Shared;
+diff --git a/code/r1/dotnet/src/EnvironmentPathRestApi.cs b/rest/src/eco.rest/Infrastructure/Economic/EnvironmentPathRestApi.cs
+similarity index 81%
+rename from code/r1/dotnet/src/EnvironmentPathRestApi.cs
+rename to rest/src/eco.rest/Infrastructure/Economic/EnvironmentPathRestApi.cs
+index 0307aa5c4c..79edcc9bfd 100644
+--- a/code/r1/dotnet/src/EnvironmentPathRestApi.cs
++++ b/rest/src/eco.rest/Infrastructure/Economic/EnvironmentPathRestApi.cs
+@@ -1,6 +1,7 @@
+-∩╗┐using Nancy;
++∩╗┐using Economic;
++using Nancy;
+ 
+-namespace Economic
++namespace eco.rest.Infrastructure.Economic
+ {
+ 	public class EnvironmentPathRestApi : EnvironmentPathBase
+ 	{
+diff --git a/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs b/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs
+index fa63fbf2d0..fced8d7b11 100644
+--- a/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs
++++ b/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs
+@@ -5,7 +5,6 @@
+ using Economic.Api.Base.Session;
+ using Economic.Rest;
+ using Economic.Rest.HttpContextWrapper;
+-using Economic.Shared;
+ using Nancy;
+ 
+ namespace eco.rest.Infrastructure.Economic
diff --git a/code/r1/dotnet/src/EnvironmentPathRestApi.cs b/rest/src/eco.rest/Infrastructure/Economic/EnvironmentPathRestApi.cs
similarity index 100%
rename from code/r1/dotnet/src/EnvironmentPathRestApi.cs
rename to rest/src/eco.rest/Infrastructure/Economic/EnvironmentPathRestApi.cs
diff --git a/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs b/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs
index fa63fbf2d0..fced8d7b11 100644
--- a/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs
+++ b/rest/src/eco.rest/Infrastructure/Economic/ServiceFactory.cs
@@ -5,7 +5,6 @@
 using Economic.Api.Base.Session;
 using Economic.Rest;
 using Economic.Rest.HttpContextWrapper;
-using Economic.Shared;
 using Nancy;
 
 namespace eco.rest.Infrastructure.Economic
diff --git a/rest/src/eco.rest/Services/CustomerService.cs b/rest/src/eco.rest/Services/CustomerService.cs
index 00d639767d..787d9eaec9 100644
--- a/rest/src/eco.rest/Services/CustomerService.cs
+++ b/rest/src/eco.rest/Services/CustomerService.cs
@@ -5,29 +5,32 @@
 using Economic;
 using Economic.Api.Base.Dto;
 using Economic.Api.Base.Entities;
+using Economic.ApplicationService.Interface;
 using Economic.Entities;
 using Economic.EntityRepositories;
 using Economic.Helpers;
 using Economic.Infrastructure.Interface;
+using Economic.Services.Sproom;
+using System.Linq;
 
-namespace eco.rest.Services
+namespace eco.rest.Services
 {
 	public interface ICustomerFutureService : ICustomerController { }
 
 	public interface ICustomerService
 	{
-		DebtorDto Get(int customerId);
-
-		void Update(EContext context, DebtorDto customer, int originalCustomerNumber);
+		DebtorDto Get(int customerId);
+
+		void Update(EContext context, DebtorDto customer, int originalCustomerNumber);
 	}
 
-	public class CustomerService : LegacyService, ICustomerService
-	{
+	public class CustomerService : LegacyService, ICustomerService
+	{
 		private readonly ICustomerRepository _customerRepository;
-		public CustomerService(IRestRequestContext ctx,
-								ICustomerRepository customerRepository) : base(ctx)
-		{
-			_customerRepository = customerRepository;
+		public CustomerService(IRestRequestContext ctx,
+								ICustomerRepository customerRepository) : base(ctx)
+		{
+			_customerRepository = customerRepository;
 		}
 
 		public DebtorDto Get(int customerId)
@@ -41,9 +44,9 @@ public DebtorDto Get(int customerId)
 				RestApiHttpStatusCodeChecks.EntityDoesNotExistErrorCheck(ex, "Customer '{0}' was not found!", customerId);
 				throw;
 			}
-		}
-
-		public void Update(EContext context, DebtorDto customerDto, int originalCustomerNumber)
+		}
+
+		public void Update(EContext context, DebtorDto customerDto, int originalCustomerNumber)
 		{
 			var currentCustomer = _customerRepository.Get(context.Connection, context.SetupId, originalCustomerNumber);
 			currentCustomer = Map(currentCustomer, customerDto);
@@ -55,14 +58,17 @@ public void Update(EContext context, DebtorDto customerDto, int originalCustomer
 				{
 					currentCustomer.Zone = convertedZone;
 				}
-			}
+			}
 			_customerRepository.Update(context, currentCustomer);
 
-			var webHook = new WebHooks_CustomerUpdated();
+			var webHook = new WebHooks_CustomerUpdated();
 			webHook.RegisterWebHookEvent_EntityUpdated(context, originalCustomerNumber.ToString(), currentCustomer.CustomerNumber.ToString());
-			context.ServiceContainer.Get<IUserLogService>().CustomerEdited(context, currentCustomer.CustomerNumber);
-		}
-
+			context.ServiceContainer.Get<IUserLogService>().CustomerEdited(context, currentCustomer.CustomerNumber);
+			
+			var changedIdentifiers = context.GetService<ICustomerApplicationService>().GetChangedIdentifiers(currentCustomer);
+			context.GetService<ISproomService>().GetCompanyElectronicInvoiceCapability(changedIdentifiers);
+		}
+
 		private DebtorDto Map(Debtor employee)
 		{
 			return new DebtorDto
@@ -98,9 +104,9 @@ private DebtorDto Map(Debtor employee)
 				OurReferenceId = employee.OurReferenceId,
 				//HandleNumber = employee.
 			};
-		}
-
-		private Customer Map(Customer currentCustomer, DebtorDto debtor)
+		}
+
+		private Customer Map(Customer currentCustomer, DebtorDto debtor)
 		{
 			currentCustomer.Name = debtor.Name;
 			currentCustomer.Address1 = debtor.Address;
@@ -131,8 +137,8 @@ private Customer Map(Customer currentCustomer, DebtorDto debtor)
 			currentCustomer.CustomerGroupNumber = debtor.DebtorGroupId ?? currentCustomer.CustomerGroupNumber;
 			currentCustomer.CustomerNumber = int.Parse(debtor.Number);
 			currentCustomer.Access = debtor.IsAccessible;
-			currentCustomer.LastUpdated = DateTime.Now;
-			return currentCustomer;
-		}
-	}
+			currentCustomer.LastUpdated = DateTime.Now;
+			return currentCustomer;
+		}
+	}
 }
